File: csharp/DVSClient/DVSClient/phone/validate/PhoneDetail.cs
using DVSClient.Common;
using DVSClient.Server.Phone;

namespace DVSClient.Phone.Validate
{
    public class PhoneDetail
    {
        public string OriginalOperatorName { get; }
        public string OriginalNetworkStatus { get; }
        public string OriginalHomeNetworkIdentity { get; }
        public string OriginalCountryPrefix { get; }
        public Country OriginalCountry { get; }
        public string OperatorName { get; }
        public string NetworkStatus { get; }
        public string HomeNetworkIdentity { get; }
        public string CountryPrefix { get; }
        public Country Country { get; }
        public string IsPorted { get; }
        public int? CacheValueDays { get; }
        public string DateCached { get; }
        public string EmailToSmsAddress { get; }
        public string EmailToMmsAddress { get; }

        public PhoneDetail(RestApiPhoneValidatePhoneDetail detail)
        {
            OriginalOperatorName = detail.OriginalOperatorName ?? string.Empty;
            OriginalNetworkStatus = detail.OriginalNetworkStatus ?? string.Empty;
            OriginalHomeNetworkIdentity = detail.OriginalHomeNetworkIdentity ?? string.Empty;
            OriginalCountryPrefix = detail.OriginalCountryPrefix ?? string.Empty;
            OriginalCountry = !string.IsNullOrEmpty(detail.OriginalCountryIso) ? Country.FromIso3(detail.OriginalCountryIso) : Country.Unknown;
            OperatorName = detail.OperatorName ?? string.Empty;
            NetworkStatus = detail.NetworkStatus ?? string.Empty;
            HomeNetworkIdentity = detail.HomeNetworkIdentity ?? string.Empty;
            CountryPrefix = detail.CountryPrefix ?? string.Empty;
            Country = !string.IsNullOrEmpty(detail.CountryIso) ? Country.FromIso3(detail.CountryIso) : Country.Unknown;
            IsPorted = detail.IsPorted ?? string.Empty;
            CacheValueDays = detail.CacheValueDays;
            DateCached = detail.DateCached ?? string.Empty;
            EmailToSmsAddress = detail.EmailToSmsAddress ?? string.Empty;
            EmailToMmsAddress = detail.EmailToMmsAddress ?? string.Empty;
        }
    }
}

File: typescript/src/phone/validate/phoneDetail.ts
import { Countries, Country } from "../../common/country";
import { RestApiPhoneValidatePhoneDetail } from "../../server/restApiPhoneValidatePhoneDetail";

export type PhoneDetail = {
    originalOperatorName: string;
    originalNetworkStatus: string;
    originalHomeNetworkIdentity: string;
    originalCountryPrefix: string;
    originalCountry: Country;
    operatorName: string;
    networkStatus: string;
    homeNetworkIdentity: string;
    countryPrefix: string;
    country: Country;
    isPorted: string;
    cacheValueDays?: number;
    dateCached: string;
    emailToSmsAddress: string;
    emailToMmsAddress: string;
};

export function restApiResponseToPhoneDetail(apiDetail: RestApiPhoneValidatePhoneDetail): PhoneDetail {

    return {
        originalOperatorName: apiDetail.original_operator_name??"",
        originalNetworkStatus: apiDetail.original_network_status??"",
        originalHomeNetworkIdentity: apiDetail.original_home_network_identity ?? "",
        originalCountryPrefix: apiDetail.original_country_prefix ?? "",
        originalCountry: apiDetail.original_country_iso ? Countries.fromIso3(apiDetail.original_country_iso) : Countries.Unknown,        
        operatorName: apiDetail.operator_name ?? "",
        networkStatus: apiDetail.network_status ?? "",
        homeNetworkIdentity: apiDetail.home_network_identity ?? "",
        countryPrefix: apiDetail.country_prefix ?? "",
        country: apiDetail.country_iso ? Countries.fromIso3(apiDetail.country_iso) : Countries.Unknown,
        isPorted: apiDetail.is_ported ?? "",
        cacheValueDays: apiDetail.cache_value_days,
        dateCached: apiDetail.date_cached ?? "",
        emailToSmsAddress: apiDetail.email_to_sms_address ?? "",
        emailToMmsAddress: apiDetail.email_to_mms_address ?? "",
    }


}